### PR TITLE
[#58, irods/irods#4083, irods/irods#5602, irods/irods#5827] GCC compatibility and housekeeping (master)

### PR DESCRIPTION
**Pull request by SwooshyCueb**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,58 +4,45 @@ find_package(IRODS 4.3.0 EXACT REQUIRED)
...
+include(IrodsCXXCompiler)
+set(CMAKE_CXX_STANDARD ${IRODS_CXX_STANDARD})
+set(CMAKE_MODULE_LINKER_FLAGS_INIT "-Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
+set(CMAKE_SHARED_LINKER_FLAGS_INIT "-Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,defs")
+set(CMAKE_MODULE_LINKER_FLAGS_RELEASE_INIT "-Wl,--gc-sections -Wl,-z,combreloc")
+set(CMAKE_SHARED_LINKER_FLAGS_RELEASE_INIT "-Wl,--gc-sections -Wl,-z,combreloc")
+
+include(IrodsExternals)
+
+IRODS_MACRO_CHECK_DEPENDENCY_SET_FULLPATH(JANSSON jansson2.7-0)
```
Comment (korydraughn) [inline CMakeLists.txt:18]: Do we still use *jansson* in this project?

**Review by korydraughn — APPROVED** [review]
Looks good overall.

File: CMakeLists.txt (hunk 1 above)
Comment (SwooshyCueb) [reply CMakeLists.txt:18]: Yup, see #42.

**Comment by trel** [comment]
yes, please #.

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**